pkg/dockerregistry/server/quotarestrictedblobstore.go

```diff
@@ -174,3 +177,26 @@ func admitBlobWrite(ctx context.Context, repo *repository, size int64) error {
 
 	return nil
 }
+
+// admitImage checks if the size is greater than the limit range.
+func admitImage(size int64, limit corev1.LimitRangeItem) error {
+	if limit.Type != imageapi.LimitTypeImage {
+		return nil
+	}
+
+	limitQuantity, ok := limit.Max[corev1.ResourceStorage]
+	if !ok {
+		return nil
+	}
+
+	imageQuantity := resource.NewQuantity(size, resource.BinarySI)
+	if limitQuantity.Cmp(*imageQuantity) < 0 {
+		// image size is larger than the permitted limit range max size, image is forbidden
+		return newLimitExceededError(imageapi.LimitTypeImage, corev1.ResourceStorage, imageQuantity, &limitQuantity)
+	}
+	return nil
+}
+
+func newLimitExceededError(limitType corev1.LimitType, resourceName corev1.ResourceName, requested, limit *resource.Quantity) error {
```

Comment:
Why you need this separate function ?

Reply:
this was copied pretty directly from origin where it is called twice.  I can collapse it here if it bothers you.

Reply:
Yes, please